src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -33,6 +33,11 @@ namespace Microsoft.CodeAnalysis
     public abstract partial class Compilation
     {
         /// <summary>
+        /// For use only in <see cref="Microsoft.CodeAnalysis.Collections.CompilationContext"/>.
+        /// </summary>
+        internal ConcurrentDictionary<object, object> _compilationContext = new ConcurrentDictionary<object, object>();
```

Comment:
My original comment is gone now, but I'll re-iterate: Make this a lazily allocated dictionary. An empty ConcurrentDictionary carries quite a bit of overhead.

The canonical pattern is:
```C#
private ConcurrentDictionary<object, object> _compilationContext;
internal ConcurrentDictionary<object, object> CompilationContext => LazyInitializer.EnsureInitialized(ref _compilationContext);
```

Another way to mitigate the allocation hit for an empty ConcurrentDictionary is to specify a smaller concurrency level (the default is 4 times the processor count) and initial capacity (default is 31).